### PR TITLE
Some tiny re-organizing of the code + solving the ignored tests

### PR DESCRIPTION
**Pull request by omid**
Most ignored tests, based on RFC, actually should return Error (panic in our test env) and most of the tests were useless/duplicated!
So I removed several of them and also test if the code actually return proper error or not.

Also moved the NWeekday conversions into the NWeekday file.

So after this (and the other PR) there will be no `ignored` tests.

### CODE REVIEW

**Comment by fmeringdal** [comment]
Seems like some funky git stuff happened when this PR #46 got merged. I removed all of the ignored tests in the rrule file in the previous PR #52. Somehow they got back in #46 and I don't understand how I didn't catch that before merging. I remember that you had to fix a lot of merge conflicts in #46, because it was open for a long time (which is my bad). That is probably the origin of the duplicated tests.

What I did in #52 was to delete all of the ignored tests and create a [new test in the validator](https://github.com/fmeringdal/rust-rrule/blob/main/rrule/src/validator/validate_properties.rs#L476) which rejects invalid `freq` and `by*` combos. I think all of the ignored tests can be deleted (instead of having a `should_panic`) because these things are already tested in the validator.

**Comment by omid** [comment]
Oh, nice, sorry, I mixed and missed those 🙈

File: rrule/tests/rrule.rs [hunk 1]
```diff
@@ -4226,113 +3323,8 @@ fn secondly_by_month_and_yearday() {
     );
 }
 
+#[cfg(feature = "no-validation-limits")]
```
Comment (omid) [inline rrule/tests/rrule.rs:3326]: What about these `no-validation-limits` tests @fmeringdal ?

Comment (fmeringdal) [reply rrule/tests/rrule.rs:3326]: I think they should stay. Any reasons we would want to ignore/remove them?

Comment (omid) [reply rrule/tests/rrule.rs:3326]: No, because I think you removed them in your PR #52

Comment (fmeringdal) [reply rrule/tests/rrule.rs:3326]: Maybe I am blind, but I can't see that I removed these tests. I can't think of any reasons to remove them, so let's keep them.

**Comment by fmeringdal** [comment]
I think this is good to merge. @omid let me know if want to make any more changes before merging it

**Comment by omid** [comment]
Thanks @fmeringdal yes, I think I'm done

**Pull request merged**